azure-ts-msi-keyvault-rbac/index.ts

```diff
@@ -0,0 +1,170 @@
+    storageAccountName: storageAccount.name,
+    containerAccessType: "private",
+});
+
+// Azure SQL Server that we want to access from the application
+const sqlServer = new azure.sql.SqlServer("sqlserver", {
+    resourceGroupName: resourceGroup.name,
+
+    administratorLogin: "manualadmin",
+    administratorLoginPassword: "WeWontUseTh1sPwd",
+    version: "12.0",
+});
+
+// Azure SQL Database that we want to access from the application
+const database = new azure.sql.Database("sqldb", {
+    resourceGroupName: resourceGroup.name,
+    serverName: sqlServer.name,
+    requestedServiceObjectiveName: "S0"
+});
+
+// The connection string that has no credentials in it: authertication will come through MSI
+const connectionString = pulumi.all([sqlServer.name, database.name]).apply(([server, db]) => 
+    `Server=tcp:${server}.database.windows.net;Database=${db};`);
+
+// A file in Blob Storage that we want to access from the application
+const textBlob = new azure.storage.Blob("text", {
+    resourceGroupName: resourceGroup.name,
+    storageAccountName: storageAccount.name,
+    storageContainerName: storageContainer.name,
+    type: "block",
+    source: "./README.md",
+});
+
+// A plan to host the App Service
+const appServicePlan = new azure.appservice.Plan("asp", {
+    resourceGroupName: resourceGroup.name,
+    kind: "App",
+    sku: {
+        tier: "Basic",
+        size: "B1",
+    },
+});
+
+// ASP.NET deployment package
+const blob = new azure.storage.ZipBlob("zip", {
+    resourceGroupName: resourceGroup.name,
+    storageAccountName: storageAccount.name,
+    storageContainerName: storageContainer.name,
+    type: "block",
+
+    content: new pulumi.asset.FileArchive("./webapp/bin/Debug/netcoreapp2.2/publish")
+});
+
+const clientConfig = pulumi.output(azure.core.getClientConfig({}));
+const tenantId = clientConfig.apply(c => c.tenantId);
+
+// Determine the ID of the current user or service principal
+const currentPrincipal = clientConfig.apply(c => 
+    c.servicePrincipalObjectId !== "" 
+    // Current, only service principal ID is available in the context
+    ? c.servicePrincipalObjectId
+    // If logged in with a user, find their ID via Azure CLI
+    : <string>JSON.parse(execSync("az ad signed-in-user show --query objectId").toString()));
+
+// Key Vault to store secrets (e.g. Blob URL with SAS)
+const vault = new azure.keyvault.KeyVault("vault", {
+    resourceGroupName: resourceGroup.name,
+    sku: {
+        name: "standard",
+    },
+    tenantId: tenantId,
+    // It won't setup any access if pulumi is executed under user account
+    accessPolicies: [{
+        tenantId,
+        objectId: currentPrincipal,
+        secretPermissions: ["delete", "get", "list", "set"],
+    }]        
+});
+
+// Put the URL of the zip Blob to KV
+const secret = new azure.keyvault.Secret("deployment-zip", {
+    keyVaultId: vault.id,
+    value: azure.storage.signedBlobReadUrl(blob, storageAccount),
```

Comment:
Do you know if this `value` is visibile in plain text in the checkpoint?  It might be nice to wrap `signedBlobReadUrl` in `pulumi.secret` (or whatever the API call is to turn an output into a secret), so that this gets encrypted in the checkpoint.  In general, it might be nice to double check that the sensitive values here are not replicated into the checkpoint if possible - it's not strictly critical for this example - but given this example is about trying to be more careful with access to secrets, also having it be the case that no secrets leak into the checkpoint would be a nice addition.

Reply:
Yes, it is visible in `pulumi stack export`. However, lots of things are there, e.g. for each storage account there is a `primaryAccessKey` sitting right there, or for an App Service 
```
"siteCredential": {
     "password": "redacted",
     "username": "$app902e9f6a"
},
```
The random string serving as password from the above is visible too.

Given that, the only reason to use `pulumi.secret` here seems to be to educate people it exists?

Reply:
Got it - makes sense - will be worth looking into those more in the future independent of this PR.